packages/resilience/src/retry.spec.ts

```diff
@@ -0,0 +1,97 @@
+import axios from 'axios';
+import nock from 'nock';
+import { executeWithMiddleware } from './middleware';
+import { retry } from './retry';
+
+describe('retry', () => {
+  const HTTP_STATUS = {
+    OK: 200,
+    UNAUTHORIZED: 401,
+    FORBIDDEN: 403,
+    SERVICE_UNAVAILABLE: 504
+  };
+
+  it('No retry needed', async () => {
+    nock('https://example.com', {}).get('/retry').reply(HTTP_STATUS.OK);
+
+    const request = () =>
+      axios.request({
+        baseURL: 'https://example.com',
+        method: 'get',
+        url: '/retry'
+      });
+
+    await expect(
+      executeWithMiddleware(
+        [retry(0)],
+        { uri: 'https://example.com', args: [] },
+        request
+      )
+    ).resolves.not.toThrow();
+  });
+
+  it('Needs to retry twice', async () => {
+    nock('https://example.com', {})
+      .get('/retry')
+      .reply(HTTP_STATUS.SERVICE_UNAVAILABLE)
+      .get('/retry')
+      .reply(HTTP_STATUS.SERVICE_UNAVAILABLE)
+      .get('/retry')
+      .reply(HTTP_STATUS.OK);
+
+    const request = () =>
+      axios.request({
+        baseURL: 'https://example.com',
+        method: 'get',
+        url: '/retry'
+      });
+
+    await expect(
+      executeWithMiddleware(
+        [retry(2)],
+        { uri: 'https://example.com', args: [] },
+        request
+      )
+    ).resolves.not.toThrow();
+  });
+
+  it('Does not retry on HTTP Status 401', async () => {
```

Comment:
[req] Add a second nock with status 200 after the unauthorized to ensure that no retry is done.